### PR TITLE
refactor: upgrade koa-session, support egg2. [BREAKING CHANGE]

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

drop node <8 support, so it is a breaking change.

https://github.com/eggjs/egg/issues/1564

### CODE REVIEW

**Comment by atian25** [comment]
这个库没有开 win 的测试？

**Comment by dead-horse** [comment]
依赖了 redis-server

File: .autod.conf.js [hunk 1]
```diff
@@ -8,7 +8,6 @@ module.exports = {
     'benchmark',
   ],
   devdep: [
-    'egg',
```
Comment (atian25) [inline .autod.conf.js]: 干掉后 autod 会被干掉依赖吧？

Comment (dead-horse) [reply .autod.conf.js]: 不会，dev 不会去除，因为现在依赖的是 next tag，所以就没必要在这里写了

**Comment by atian25** [comment]
回头 egg guide docs 那边也要改改对应的 generator

**Pull request merged**